doc/nrf/libraries/modem/nrf_modem_lib.rst

```diff
@@ -100,11 +100,96 @@ Thread-based processing
-The thread reads from the FIFO queue and forwards the trace data to the configured trace transport medium.
+The thread reads from the FIFO queue and forwards the trace data to the configured trace backend.
 The FIFO queue used by the modem trace module uses dedicated heap memory.
-If the modem trace medium is unable to keep up with the modem traces, the heap size can be increased by enabling :kconfig:option:`CONFIG_NRF_MODEM_LIB_TRACE_HEAP_SIZE_OVERRIDE` and configuring :kconfig:option:`CONFIG_NRF_MODEM_LIB_TRACE_HEAP_SIZE`.
-Increasing the heap size allows more traces in the FIFO queue, but the trace heap will still be depleted if the modem continues to send traces at a rate faster than the rate at which the medium can handle over time.
-If increasing the trace heap size does not help, either optimize the medium speed or use a faster trace transport medium.
+If the modem trace backend is unable to keep up with the modem traces, the heap size can be increased by enabling :kconfig:option:`CONFIG_NRF_MODEM_LIB_TRACE_HEAP_SIZE_OVERRIDE` and configuring :kconfig:option:`CONFIG_NRF_MODEM_LIB_TRACE_HEAP_SIZE`.
+Increasing the heap size allows more traces in the FIFO queue, but the trace heap will still be depleted if the modem continues to send traces at a rate faster than the rate at which the backend can handle over time.
+If increasing the trace heap size does not help, either optimize the backend speed or use a faster trace backend.
+
+.. _adding_custom_modem_trace_backends:
+
+Adding custom trace backends
+=====================================
+
+You can add a custom trace backends if the existing trace backends are not sufficient.
+At any time, only one trace backend can be compiled with the application.
+The value of the kconfig:option:`CONFIG_NRF_MODEM_LIB_TRACE_BACKEND` Kconfig option determines which trace backend is compiled.
+The :ref:`modem_trace_backend` sample demonstrates how a custom trace backend can be added to an application.
+
+Complete the following steps to add a custom trace backend:
+
+1. Place the files that have the custom trace backend implementation in a library or an application you create.
+For example, the implementation of the UART trace backend (default) can be found in the :file:`nrf/lib/nrf_modem_lib/trace_backends/uart/uart.c` file.
+
+2. Add a C file implementing the interface declared in the :file:`nrf/include/modem/trace_backend.h` header file.
+
+   .. code-block:: c
+
+   /* my_trace_backend.c */
+
+   #include <modem/trace_backend.h>
+
+   int trace_backend_init(void)
+   {
+           /* initialize trace backend here */
+           return 0;
+   }
+
+   int trace_backend_deinit(void)
+   {
+           /* optional deinitialization code here */
+           return 0;
+   }
+
+   int trace_backend_write(const void *data, size_t len)
+   {
+           /* forward data over custom backend here */
+           /* return number of bytes written or negative error code on failure */
+           return 0;
+   }
+
```

Comment:
```suggestion
#. Add a C file implementing the interface in :file:`nrf/include/modem/trace_medium.h` header file.
   
   .. code-block:: c
      
      /* my_trace_medium.c */
      #include <modem/trace_medium.h>
      int trace_medium_init(void)
      {
           /* initialize transport medium here */
           return 0;
      }
      int trace_medium_deinit(void)
      {
           /* optional deinitialization code here */
           return 0;
      }
      int trace_medium_write(const void *data, size_t len)
      {
           /* forward data over custom transport here */
           /* return number of bytes written or negative error code on failure */
      }
```
This should fix the doc build error. The code written should be in line with 'c' of the 'code-block'.